File: src/main/java/com/wesker926/leetcode/algorithms/p0495/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0495.s1;

/**
 * @author wesker.gh
 * @date 2021/11/10
 * @description 遍历
 */
public class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int sum = 0;
        for (int i = 0; i < timeSeries.length - 1; i++) {
            sum += Math.min(timeSeries[i + 1] - timeSeries[i], duration);
        }
        return sum + duration;
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0495/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/10
// @description: 遍历
package s1

func findPoisonedDuration(timeSeries []int, duration int) int {
	sum := 0
	for i := 0; i < len(timeSeries)-1; i++ {
		sum += minInt(timeSeries[i+1]-timeSeries[i], duration)
	}
	return sum + duration
}

func minInt(i, j int) int {
	if i <= j {
		return i
	}
	return j
}
